### DISCUSSION TITLE
How can I fix TS error "Types of property 'onDone' are incompatible" while using onDone: send('AN_EVENT_NAME')

### DISCUSSION THREAD

**Discussion by mogarick** [statelyai/xstate#2853] [Q&A] [answered]
Hi, 

Is there a a way to fix this TS error when using `send` as action creator for onDone when invoking a Promise?

Here is my codesandbox:

https://stately.ai/viz/7e762ff6-2cc0-4ccb-9e1b-a22383b3803c

**Comment by mattpocock** [comment, marked as answer]
`onDone: { actions: send() }`

**Reply by mogarick** [reply]
Hi @mattpocock,
 thank you for your quick attention but it doesn't work :(

I've updated the [sandbox](https://stately.ai/viz/7e762ff6-2cc0-4ccb-9e1b-a22383b3803c) to 

```
onDone: {actions: send("MY_AVENT_NAME")}
```

but I'm still getting an error:

```
Types of property 'actions' are incompatible.
                        Type 'SendAction<unknown, EventObject, EventObject>' is not assignable to type 'Actions<SecurityServiceSMachineContext, DoneInvokeEvent<any>> | undefined'.
                          Type 'SendAction<unknown, EventObject, EventObject>' is not assignable to type 'string'.(2345)
```

Any other idea or suggestion?

Thank you in advance.

PD: It's important to note that the code works. Only the TS checks are failing.

**Reply by mattpocock** [reply]
Working fine on this forked version with the position of the variables tweaked:

https://stately.ai/viz/5f899b38-9ca9-420a-96a1-f22b7b489370

**Reply by mogarick** [reply]
@mattpocock I see and tried it on my version! 

Thank you very much!

So do you recommend to inline the MachineConfig object directly into the createMachine call to avoid such kind of TS errors? if so, why would this tweak be needed?
